gossip/minimal/gossip.go

```diff
@@ -533,12 +534,25 @@ func (src *sourceLog) GetNewerEntries(ctx context.Context, g *Gossiper, lastSTH,
 	prevTreeSize := lastSTH.TreeSize
 	glog.V(1).Infof("Retriever(%s): Previous Tree Size (%v)", src.Name, prevTreeSize)
 
-	startIndex, endIndex := int64(prevTreeSize+1), int64(prevTreeSize+newTreeSize)
+	startIndex, endIndex := int64(prevTreeSize+1), int64(newTreeSize)
 	glog.V(1).Infof("Get newer entries for source log %s from (%v) to (%v)", src.Name, startIndex, endIndex)
 	entries, err := src.Log.GetEntries(ctx, startIndex, endIndex)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get new entries: %v", err)
 	}
-
+	proof, err := src.Log.GetSTHConsistency(ctx,uint64(startIndex),uint64(endIndex))
```

Comment:
I think we should do this verification should be explicitly invoked by another function --- the one that currently consumes/processes entries returned by this function. Let's write functions for a single purpose, for as much as we can.